crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -245,6 +245,15 @@ impl BatchJob {
             })
             .await
             .map_err(BuildBatchError::JoinError)?,
+        }?;
+
+        if proven_batch.proof_security_level() < MIN_PROOF_SECURITY_LEVEL {
+            Err(BuildBatchError::SecurityLevelTooLow(
+                proven_batch.proof_security_level(),
+                MIN_PROOF_SECURITY_LEVEL,
+            ))
+        } else {
+            Ok(proven_batch)
         }
```

Comment:
Should we make this a bit more flexible and not hard-code the minimum security level? Basically, should we make `min_proof_security` be a property of `BatchBuilder` or maybe even `BatchProver`?

Reply:
I'm not sure. It would make sense to do that if the security level needs to be higher than the configured prover (assuming it can change). But if we're only interested in it being higher than the minimum (which is what we're currently doing), the current check would be fine.

Reply:
I think for now this is fine as is. We may want to make security level a config parameter of the node - but that would be a separate task. Let's create an issue for this.